src/librustc/traits/select.rs

```diff
@@ -2495,6 +2547,40 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
         Ok(VtableFnPointerData { fn_ty: self_ty, nested: obligations })
     }
 
+    fn confirm_generator_candidate(&mut self,
```

Comment:
`probe.rs` implementing a hacked-together version of trait selection is just a crazy thing that should be fixed. Chalk might make `select.rs`/`project.rs` simpler.